### ISSUE TITLE
Preview feature feedback: MongoDB support (`mongoDb`)

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#8241]
Please share your feedback about the `mongoDb` functionality released in [v2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) in this issue.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo.
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this issue.

If you have any questions, don't hesitate to ask them in the [`#prisma-client`](https://prisma.slack.com/archives/CCWDULGUW) channel in the [Prisma Slack](https://slack.prisma.io).

**Comment by timleslie** [comment]
https://github.com/prisma/prisma/issues/8261

**Comment by hinsxd** [comment]
https://github.com/prisma/prisma/issues/8389
This occurs only on v2.27 and v2.28

**Comment by pkellner** [comment]
I love this.  Very important IMHO. I'm wondering why it is called "Interactive Transaction" and not just "Transaction".  What is interactive about it?

**Comment by Lefcott** [comment]
Hi! I think it would be great to have a way to model nested fields, since it's something quite common when using MongoDB.

At this moment, I would have to define two models, for example:

```prisma
model Match {
  id        String     @id @default(dbgenerated()) @map("_id") @db.ObjectId
  movements Movement[]
}

model Movement {
  id          String     @id @default(dbgenerated()) @map("_id") @db.ObjectId
  match       UserMatch? @relation(fields: [matchId], references: [id])
  matchId     String?    @db.ObjectId
  date        DateTime
}
```

And it would be great to have something like this:

```prisma
model Match {
  id        String     @id @default(dbgenerated()) @map("_id") @db.ObjectId
  movements [
    {
      date DateTime
    }
  ]
}
```

Hope this is useful, thanks!

**Comment by 2color** [comment]
Hey @Lefcott,

Thanks for sharing your feedback. We have an [open issue for embedded documents](https://github.com/prisma/prisma/issues/6708) which would allow you to type nested objects. Feel free to give it a thumbs up and share any thoughts you have on the design proposal.

**Comment by jbeckton** [comment]
So is introspections supported or not? The getting started tutorial has you go through the steps of introspecting an existing MongoDB but on the MongoDB connector page is says it's not supported. Is there at least some support for introspection for MongoDB?

**Comment by janpio** [comment]
Introspection for MongoDB was added to the `mongodb` preview feature with 3.2.0: https://github.com/prisma/prisma/releases/tag/3.2.0 Can you point where we have outdated information in our docs - or even send a PR fixing it please?

**Comment by jbeckton** [comment]
https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#introspection-and-migration

**Comment by janpio** [comment]
https://github.com/prisma/docs/pull/2781

**Comment by Skellord** [comment]
Hi! I have existing DB, which have collections, created by import from TSV files. Now i  just trying to connect Prisma and there is the question. Can i provide collection name, when i describing model in schema? Or i need to describe all fields in model, to make query working. For now i describe few fields in model, but existing mongo document have more fields, and Prisma query doesn't work. Maybe i miss something?

**Comment by Jolg42** [comment]
@Skellord so here if I understand, you only specified some fields in your model in the Prisma schema, and you are trying to query using the Prisma Client, but it fails?
Which query are you running?
What error do you get?

I can recommend introspecting the collection using `db pull`, which will create a Prisma schema, see 
https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/mongodb/introspection-typescript-mongodb#introspecting-mongodb-with-prisma

**Comment by Skellord** [comment]
@Jolg42 thanks for answering. Yes, you got me right, and yes, i miss introspecting step -_-. Now it's working. Thank you again, and sorry for wasting time.

**Comment by matthewmueller** [comment]
MongoDB General Availability shipped in [3.12.0](https://github.com/prisma/prisma/releases/3.12.0)!

If you run into any issues, please open a new issue. Thanks everyone!
